UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneExporter.cs

```diff
@@ -529,43 +604,41 @@ private PbrMetallicRoughness ExportPBRMetallicRoughness(UnityEngine.Material mat
 
 				if (mainTex != null)
 				{
-					pbr.BaseColorTexture = ExportTextureInfo(mainTex);
+					pbr.BaseColorTexture = ExportTextureInfo(mainTex, TextureMapType.Main);
 					ExportTextureTransform(pbr.BaseColorTexture, material, "_MainTex");
 				}
 			}
 
 			if (material.HasProperty("_Metallic"))
 			{
-				pbr.MetallicFactor = material.GetFloat("_Metallic");
+                var metallicGlossMap = material.GetTexture("_MetallicGlossMap");
+                pbr.MetallicFactor = (metallicGlossMap != null) ? 1.0 : material.GetFloat("_Metallic");
 			}
 
-			if (material.HasProperty("_Roughness"))
```

Comment:
newer versions of Unity uses _Roughness instead. I think you will break that code if you make this change.

Reply:
Not sure what happened there, will fix it.

Reply:
Update: since I integrated PR #133 with this one, it pulled changes in that PR too. I think @justinctlam might have some explanation as to why he took out the _Roughness factor

Reply:
Which Unity uses _Roughness? I am on Unity 2017.3.1f1. I looked at the Standard.shader and StandardRoughness.shader and I don't see the property. Can you point me to it?

Reply:
You are right. This is my mistake. So, this never worked before then, right?